### PR TITLE
Java: Promote XSLT Injection from experimental

### PR DESCRIPTION
**Pull request by atorralba**
PR to promote the XSLT Injection query created in #3363

## Changes

* Existing files were moved out of experimental
* The `XsltInjectionLib.qll` file was renamed and refactored to use the CSV sink model.
* Added tests using `InlineExpectationsTest`.

## Evaluation

[CVE-2012-1592](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2012-1592) is correctly detected after adding ad hoc additional taint steps for `javax.servlet.ServletConext.getResource` and `java.net.URL.openStream`. Considering adding the later as a flow summary in a separate PR (because of the impact it'll probably have in several queries).

### CODE REVIEW

**Comment by atorralba** [comment]
@github/docs-content-codeql please review the qhelp file. Even though changes aren't introduced in this PR, it wasn't reviewed when this query was merged to experimental.

File: java/ql/src/Security/CWE/CWE-074/XsltInjection.ql [hunk 1]
```diff
@@ -11,8 +11,7 @@
  */
 
 import java
-import semmle.code.java.dataflow.FlowSources
-import XsltInjectionLib
+import semmle.code.java.security.XsltInjectionQuery
```
Comment (mchammer01) [inline java/ql/src/Security/CWE/CWE-074/XsltInjection.ql:14]: Comment for line 3 above (as I can't comment on the line itself):
Suggesting:
`Performing an XSLT transformation with user-controlled stylesheets can lead to
information disclosure or execution of arbitrary code.`

**Review by mchammer01 — COMMENTED** [review]
@atorralba - this LGTM ✨ 
I'm making a minor suggestion to the query description + I fixed a few minor things in the qlhep file (hope it' ok for me to directly commit to your branch as I don't appear to be able to make suggestions to that file).

File: java/ql/src/Security/CWE/CWE-074/XsltInjection.ql (hunk 1 above)
Comment (atorralba) [reply java/ql/src/Security/CWE/CWE-074/XsltInjection.ql:14]: Don't worry about directly committing to the branch, It's totally fine :-)

Thank you for the review! I applied your suggestion regarding the query description.

**Comment by aschackmull** [comment]
LGTM.

**Comment by atorralba** [comment]
Rebased to fix conflicts.

**Review by aschackmull — APPROVED** [review]

**Pull request merged**